keccak256/src/permutation/tables.rs

```diff
@@ -486,202 +489,257 @@ impl<F: Field> Base13toBase9TableConfig<F> {
+                    let output_b13 = f_from_radix_be::<F>(&b2_chunks, B13);
+
+                    self.map.insert(input_b2.to_repr(), (output_b9, output_b13));
+                    // Iterate over all possible binary values of size 16
+
                     table.assign_cell(
-                        || "base 9",
-                        self.base9,
+                        || "base 2",
+                        self.lookup_config.col0.1,
                         i,
-                        || Ok(f_from_radix_be::<F>(&b9_chunks, B9)),
+                        || Ok(input_b2),
                     )?;
-                    let converted_chunks: Vec<u8> =
-                        b9_chunks.iter().map(|&x| convert_b9_coef(x)).collect_vec();
+
                     table.assign_cell(
-                        || "base 13",
-                        self.base13,
+                        || "base 9",
+                        self.lookup_config.col1.1,
                         i,
-                        || Ok(f_from_radix_be::<F>(&converted_chunks, B13)),
+                        || Ok(output_b9),
                     )?;
                     table.assign_cell(
-                        || "base 2",
-                        self.base2,
+                        || "base 13",
+                        self.lookup_config.col2.1,
                         i,
-                        || Ok(f_from_radix_be::<F>(&converted_chunks, B2)),
+                        || Ok(output_b13),
                     )?;
                 }
                 Ok(())
             },
         )
     }
 
-    pub fn configure(meta: &mut ConstraintSystem<F>) -> Self {
-        Self {
-            base2: meta.lookup_table_column(),
-            base9: meta.lookup_table_column(),
-            base13: meta.lookup_table_column(),
-            _marker: PhantomData,
-        }
+    pub fn configure(
+        meta: &mut ConstraintSystem<F>,
+        adv_cols: [Column<Advice>; 3],
+        table_cols: [TableColumn; 3],
+    ) -> Self {
+        let lookup_config = ThreeColumnsLookup::configure(meta, adv_cols, table_cols, "from base9");
+        let map = HashMap::new();
+        Self { lookup_config, map }
     }
+    pub fn assign_region(
+        &self,
+        layouter: &mut impl Layouter<F>,
+        input: &AssignedCell<F, F>,
+    ) -> Result<
+        (
+            Vec<AssignedCell<F, F>>,
+            Vec<AssignedCell<F, F>>,
+            Vec<AssignedCell<F, F>>,
+        ),
+        Error,
+    > {
+        let input_coefs = compute_input_coefs::<F, NUM_OF_BINARY_SLICES>(
+            input.value(),
+            B2,
+            NUM_OF_BINARY_CHUNKS_PER_SLICE,
+        );
+        layouter.assign_region(
+            || "base 2",
+            |mut region| {
+                let mut input_cells = vec![];
+                let mut output_b9_cells = vec![];
+                let mut output_b13_cells = vec![];
+                for (offset, input_coef) in input_coefs.iter().enumerate() {
```

Comment:
This is where I think we can optimize this a way more by not increasing the offset each time we add and input and instead increase the amount of columns used and not repeat it.